src/doc/The_Configuration_File.md

```diff
@@ -1,147 +1,184 @@
 Configuration File
 ==================
 
```

Comment:
I put together my proposed changes to this file in https://github.com/hammerlab/ketrew/tree/json_config_the_doc. Some are stylistic/wording choices so not necessary to merge the entire thing, do as you see fit.